tests/functional/TestSmokeTest/tests/timeout_test.go

```diff
@@ -0,0 +1,102 @@
+// +build functional
+
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+package functional
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/heketi/pkg/remoteexec/ssh"
+	"github.com/heketi/heketi/pkg/testutils"
+
+	"github.com/heketi/tests"
+)
+
+var (
+	nodes     = 4
+	disk      = 4
+	pidFile   = "/var/tmp/hook.pid"
+	delayFile = "/var/tmp/hook-delay"
+	killPid   = "/var/tmp/kill-pid.sh"
+)
+
+func addDelay(timeOut int, node string, exec *ssh.SshExec) error {
+	dumpData := fmt.Sprintf("echo -e %dm >%s", timeOut, delayFile)
+	cmd := []string{dumpData}
+	_, err := exec.ConnectAndExec(node, cmd, 10, true)
+	return err
+}
+
+func cleanDelay(node string, exec *ssh.SshExec) error {
+	removeSleep := fmt.Sprintf("sh %s", killPid)
+	removeDelayFile := fmt.Sprintf("rm -f %s", delayFile)
+	removePIDFile := fmt.Sprintf("rm -f %s", pidFile)
+
+	cmd := []string{removeSleep, removeDelayFile, removePIDFile}
+	_, err := exec.ConnectAndExec(node, cmd, 10, true)
+	return err
+}
+
+func cleanupOp(exec *ssh.SshExec, t *testing.T) {
+	for i := 0; i < nodes; i++ {
+		sshHost := cenv.SshHost(i)
+		err := cleanDelay(sshHost, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+}
+func TestGlusterTimeout(t *testing.T) {
+	// Setup the VM storage topology
+	teardownCluster(t)
+	setupCluster(t, nodes, disk)
+	defer teardownCluster(t)
+
+	na := testutils.RequireNodeAccess(t)
+	//add delay to postscript in all gluster nodes
+	exec := na.Use(logger)
+	for i := 0; i < nodes; i++ {
+		err := addDelay(4, cenv.SshHost(i), exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+
+	//cleanupOp deletes the delay file and stops the sleep process
+	//which was started as part of pre script
+	ct := time.AfterFunc(3*time.Minute, func() {
+		cleanupOp(exec, t)
+	})
+	defer ct.Stop()
```

Comment:
I'm a bit confused by this section. First off I don't understand the need to do the cleanup in the background.
Second, I am concerned by the length of the sleep. Three minutes is quite a long time for something in a test suite that is already taking far too long to run. :-)
Because the functional test system is generally working and gluster commands complete reasonably quickly I think we ought to work on the scale of tens-of-seconds, not minutes.